glean-core/preview/src/lib.rs

```diff
@@ -75,46 +82,65 @@ where
     F: Fn(&Glean) -> R,
 {
     let lock = global_glean().lock().unwrap();
-    f(&lock)
+    f(&lock.instance)
+}
+
+fn with_glean_wrapper_mut<F, R>(f: F) -> R
+where
+    F: Fn(&mut GleanWrapper) -> R,
+{
+    let mut lock = global_glean().lock().unwrap();
+    f(&mut lock)
 }
 
 fn with_glean_mut<F, R>(f: F) -> R
 where
     F: Fn(&mut Glean) -> R,
 {
     let mut lock = global_glean().lock().unwrap();
-    f(&mut lock)
+    f(&mut lock.instance)
 }
 
 /// Create and initialize a new Glean object.
 ///
 /// See `glean_core::Glean::new`.
 pub fn initialize(cfg: Configuration, client_info: ClientInfoMetrics) -> Result<()> {
-    let channel = cfg.channel;
-    let cfg = glean_core::Configuration {
+    let core_cfg = glean_core::Configuration {
         upload_enabled: cfg.upload_enabled,
-        data_path: cfg.data_path,
-        application_id: cfg.application_id,
+        data_path: cfg.data_path.clone(),
+        application_id: cfg.application_id.clone(),
         max_events: cfg.max_events,
         delay_ping_lifetime_io: cfg.delay_ping_lifetime_io,
     };
```

Comment:
There's no adorably-short rustism for partially initing from another instance? Some sort of 
```rust
let core_cfg = glean_core::Configuration{data_path: cfg.data_path.clone(), ..cfg}; ?
```

Ah-ha, I knew I knew it from somewhere. [Struct update syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) should work here, right?

Reply:
That only works if its of the same type (and cfg is a `glean_preview::Configuration`, not a `glean_core::Configuration`).

I was slightly annoyed by this as well, but given that it's all internal, it didn't bother me enough.